azurerm/resource_arm_storage_account.go

```diff
@@ -495,6 +500,11 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 	}
 	d.Set("account_kind", resp.Kind)
 
+	log.Printf("[INFO] Identity is %q", resp.Identity)
+	if identity := resp.Identity; identity != nil {
```

Comment:
@lstolyarov you'll want to use `resp.[XX]Properties.Identity` here - since the top level fields `resp.Identity` isn't guaranteed to have a value, unfortunately. These fields exist primarily for the older API's where the responses aren't guaranteed in the `properties` block in the JSON

Reply:
@tombuildsstuff what do you mean by `[XX]Properties`? There is an `AccountProperties` filed in `resp` but that does not have an `Identity` field

Reply:
>  @tombuildsstuff what do you mean by `[XX]Properties`? There is an `AccountProperties` filed in `resp` but that does not have an `Identity` field 

The `properties` field is prefixed with the name of the struct, so it's different (hence `[XX]Properties`, sorry I should have looked this up).

I don't see any option to enable this in the Portal - is this feature in Preview?

Reply:
Well it is not really a feature. This PR is to be able to extract the Object ID (or Principal ID as it is referred to in the SDK). I don't think there is any anyway to get this information out in the portal - it could be extracted with powershell.

Reply:
sorry, I was referring to pulling this out via the API/SDK - I'm not sure if you've seen it but [the Azure Resources Explorer](https://resources.azure.com) can be really helpful here.

In other API's (e.g. App Service) the `identity` block won't be returned from the API until it's enabled, which can be done by sending the following Request (in the Create/Update):

```
Identity: &storage.Identity{
  Type: utils.String("SystemAssigned"),
},
```

Once the Identity's assigned, the `identity` block returned from the API SDK as Principal ID/Object ID. That said, given this isn't available in the Portal yet - and I can't see any reference to it online; I have a feeling this may be in Preview?

Reply:
Ok, that makes more sense now. 

Thanks for the Azure Resources Explorer - its really useful.

I am not sure if it would technically count as a preview feature or not. My understanding that behind the scenes all resources have object ids and this is required as input for key vault. It is not really useful in the portal as when you enable storage account encryption and select the key  vault you want to use that object id gets populated automatically.